xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/macros.vm

```diff
@@ -1120,7 +1120,7 @@ $html
   #foreach($docName in $docNames)
     #if($xwiki.hasAccessLevel("view", $xcontext.user, $docName))
       #set($document = $xwiki.getDocument($docName).getTranslatedDocument())
-      #if(!$blacklistedSpaces.contains($document.getSpace()))
+      #if(("$!blacklistedSpaces" != '' && !$blacklistedSpaces.contains($document.getSpace())) || ("$!blacklistedSpaces" == '' && ($services.user.properties.displayHiddenDocuments() || !$document.isHidden())))
```

Comment:
I understand the desire to keep supporting $blacklistedSpaces where it's expected to be an API, but I feel that, at least for consistency, we should always do the hidden document filtering even in cases where $blacklistedSpaces is configured.

Reply:
As I said, I don't really care much about the handling of the `$blacklistedSpaces`. My concern is that I simply don't see any reason to not take into account the hidden document side of things when it's set.

Reply:
Done with my last commit.